### PR TITLE
My Portfolio Dashboards

### PR DESCRIPTION
**Pull request by rmroot**
connects #214 
connects #297 
connects #296 

- adds my portfolio page
- adds company dashboard
- adds facility dashboard
- adds assessment dashboard
- deletes old dashboard components that are no longer used
- moves forms to shared modules so can be used in dashboard and setup wizard
- add re-usable stub of services for unit tests
- minor UI/UX fixes KPI selection
- Updated styles to use consistent navigation button colors
- adds in module for pagination, implemented on latest visit table
- created folder and split up routes into own files for wizard and dashboard
- added animation module for smoother transitions between pages
- initialize new objects with names

### CODE REVIEW

File: src/app/models/contact.ts [hunk 1]
```diff
@@ -23,8 +23,8 @@ export function getNewIdbContact(userId: string, companyId: string): IdbContact
     let idbEntry: IdbEntry = getNewIdbEntry();
     return {
         ...idbEntry,
-        firstName: undefined,
-        lastName: undefined,
+        firstName: 'New',
+        lastName: 'Contact',
```
Comment (RLiNREL) [inline src/app/models/contact.ts:27]: There were previous discussions about setting default contact name, facility name, company name to empty, see #164 
Any particular reason to bring them back? "New Contact" is probably not a good default name 😄

File: src/app/setup-wizard/pre-visit/company-kpi-select/company-kpi-select.component.html [hunk 2]
```diff
@@ -6,25 +6,18 @@ <h5>
...
 <nav class="setup-wizard-footer navbar shadow">
     <div class="container-fluid justify-content-between">
-        <button class="btn btn-dark btn-sm" (click)="goBack()">
+        <button class="btn nav-btn btn-sm" (click)="goBack()">
             <fa-icon [icon]="faChevronLeft"></fa-icon>
             Go Back
         </button>
-        <button class="btn btn-dark btn-sm" (click)="goToKpiDetails()">
+        <button class="btn nav-btn btn-sm" (click)="goToKpiDetails()">
             Detail KPIs
```
Comment (RLiNREL) [inline src/app/setup-wizard/pre-visit/company-kpi-select/company-kpi-select.component.html]: perhaps, change to Facility Details when companyKpis.length is 0.
    {{ companyKpis.length > 0 ? 'Detail KPIs' : 'Facility Details' }}

File: src/app/shared/shared-company-forms/company-contacts-form/company-contacts-form.component.ts [hunk 3]
```diff
@@ -62,24 +71,28 @@ export class CompanyContactsFormComponent {
...
-  toggleBS() {
-    this.bootstrapService.bsCollapse('#' + this.contactGuid);
-    if (this.accordionGuid != this.contactGuid) {
-      this.accordionGuid = this.contactGuid;
-    } else {
-      this.accordionGuid = undefined;
-    }
-    this.localStorageDataService.setContactAccordionGuid(this.accordionGuid);
+  dislayWarningModal() {
+    this.routeGuardWarningModal = true;
```
Comment (RLiNREL) [inline src/app/shared/shared-company-forms/company-contacts-form/company-contacts-form.component.ts:92]: consider fixing the typo displayWarningModal() {}

File: src/app/setup-wizard/pre-visit/company-kpi-select/company-kpi-select.component.html (hunk 2 above)
Comment (rmroot) [reply src/app/setup-wizard/pre-visit/company-kpi-select/company-kpi-select.component.html]: Caught me being lazy. Fixed thanks

File: src/app/models/contact.ts (hunk 1 above)
Comment (rmroot) [reply src/app/models/contact.ts:27]: For this one when a user starts a new wizard they go into set company info and may not get all the way to the facility at the time. When we initialize a facility without a name, they can still navigate away and then in all other places referencing that facility it is blank. At this time we don't have good enough mechanisms in place to enforce data entry of the objects that are being left blank. We can remove the default "New ***" initialization when we have better overall validation and user messaging. In VERIFI and MEASUR we also initialize all names with "New ***", (New Assessment, New Motor, New Facility, etc.)

**Review by RLiNREL — APPROVED** [review]

**Pull request merged**